static/js/lib/validation_test.js

```diff
@@ -165,18 +165,32 @@ describe("validation library", () => {
       })
     })
 
-    it("should complain about an empty article post", () => {
-      const post = {
-        value: {
-          postType:        LINK_TYPE_ARTICLE,
-          title:           "potato",
-          article_content: []
-        }
-      }
-      assert.deepEqual(validatePostCreateForm(post), {
-        value: {
-          article_content: "Article must not be empty"
+    //
+    ;[
+      [[], true],
+      [[{ attributes: {}, children: [] }], true],
+      [[{ attributes: {}, children: [{ hey: "there" }] }], false]
+    ].forEach(([articleContent, shouldGiveError]) => {
+      it(`${shouldIf(shouldGiveError)} if articleContent = ${JSON.stringify(
+        articleContent
+      )}`, () => {
```

Comment:
Take or leave: I'd rather add another parameter describing the test case than stringifying the full JSON block